src/vs/workbench/parts/lib/node/settingsVariables.ts

```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as Types from 'vs/base/common/types';
+import { AbstractSystemVariables } from 'vs/base/common/parsers';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+
+export class SettingsVariables extends AbstractSystemVariables {
+	private configuration: any;
+	constructor(private configurationService: IConfigurationService) {
+		super();
+		this.configuration = this.configurationService.getConfiguration();
+	}
+
+	protected resolveString(value: string): string {
+		let regexp = /\$\{settings\.(.*?)\}/g;
+		return value.replace(regexp, (match: string, name: string) => {
+			let config = this.configuration;
+			let newValue = new Function('_', 'try {return _.' + name + ';} catch (ex) { return "";}')(config);
+			return Types.isString(newValue) ? newValue : '';
```

Comment:
We try to avoid using the 'new Function' pattern, is it possible to re-write this using a lambda function - it would read much nicer?